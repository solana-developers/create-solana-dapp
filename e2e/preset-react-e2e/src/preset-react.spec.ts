import { createTestProject, logger } from '@solana-developers/e2e-utils'
import { execSync } from 'child_process'
import { rmSync } from 'fs'

describe('preset-react', () => {
  const cleanUp = process.env['CLEANUP'] !== 'false'
  const packageName = '@solana-developers/preset-react'
  logger.log(`Running tests for ${packageName}`)

  let projectDirectory: string

  beforeAll(() => {
    projectDirectory = createTestProject('preset-react-e2e')

    // The plugin has been built and published to a local registry in the jest globalSetup
    // Install the plugin built with the latest source code into the test repo
    execSync(`npm install ${packageName}@e2e`, {
      cwd: projectDirectory,
      stdio: 'inherit',
      env: process.env,
    })
  })

  afterAll(() => {
    if (!cleanUp) {
      logger.log(`Skipping cleanup of test project "${projectDirectory}"`)
      return
    }
    // Cleanup the test project
    rmSync(projectDirectory, {
      recursive: true,
      force: true,
    })
    logger.log(`Cleaned up test project "${projectDirectory}"`)
  })

  it('should be installed', () => {
    // npm ls will fail if the package is not installed properly
    execSync(`npm ls ${packageName}`, {
      cwd: projectDirectory,
      stdio: 'inherit',
    })
    logger.log(`Installed ${packageName}`)
  })

  it('should be run the preset generator with only a name ', () => {
    execSync(`nx g ${packageName}:preset --name my-app --ui tailwind --anchor counter`, {
      cwd: projectDirectory,
      stdio: 'inherit',
    })
    logger.log(`Ran ${packageName}:preset`)
  })
})
