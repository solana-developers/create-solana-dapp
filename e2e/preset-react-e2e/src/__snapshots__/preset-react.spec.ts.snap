// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`preset-react should print the help command 1`] = `
"> NX generate @solana-developers/preset-react:react-application [name] [options,...]
From: @solana-developers/preset-react (v0.0.1)
Name: react-application (aliases: application, preset)
Generates a React application.
Options:
--name Name of the application [string]
--anchor Anchor template to use [string] [choices: "none",
"counter", "hello-world"]
[default: "counter"]
--anchorName Anchor project name [string] [default: "anchor"]
--port Port to run the application on [number] [default: 3000]
--ui The UI library to use [string] [choices: "none",
"tailwind"] [default:
"tailwind"]
--webName Name of the web application [string]
(overrides name)
--skipFormat Skip formatting files [boolean]"
`;
